.github/copilot-instructions.md

```diff
@@ -246,4 +246,118 @@ Located in `scripts/` directory:
+
+### Developer Documentation (`doc/` directory)
+- **[doc/README.md](../doc/README.md)**: Main developer documentation hub
+- **[doc/Configuration-Encryption.md](../doc/Configuration-Encryption.md)**: AES-256 encryption for sensitive config (API keys, tokens)
+- **[doc/Provider-Configuration-Pattern.md](../doc/Provider-Configuration-Pattern.md)**: How to create new social media providers
+- **[doc/QueueIntegration.md](../doc/QueueIntegration.md)**: Azure Queue integration for external apps
+- **[doc/Azure-KeyVault-Configuration.md](../doc/Azure-KeyVault-Configuration.md)**: Azure Key Vault setup and usage
+
+### User Documentation (`user-docs/` directory)
+- **[user-docs/Moderator_User_Manual.md](../user-docs/Moderator_User_Manual.md)**: Complete guide for content moderators
+
+### Provider-Specific Documentation
+- **[src/TagzApp.Providers.Blazot/ReadMe.md](../src/TagzApp.Providers.Blazot/ReadMe.md)**: Blazot API integration guide
+- Each provider project may contain specific README files for setup instructions
+
+## Provider System Architecture
+
+### Creating New Providers
+When adding a new social media provider:
+
+1. **Create Provider Project**: Follow naming pattern `TagzApp.Providers.[PlatformName]`
+2. **Implement Interface**: Implement `ISocialMediaProvider` from `TagzApp.Common`
+3. **Configuration UI**: Create Blazor component in `TagzApp.Blazor.Client/Components/Admin/[Platform].Config.Ui.razor`
+4. **Reference Documentation**: See `doc/Provider-Configuration-Pattern.md` for complete pattern
+5. **Add Tests**: Include unit tests in `TagzApp.UnitTest` project
+
+### Provider Configuration Pattern
+All providers follow a consistent pattern:
+- Configuration stored in database (optionally encrypted)
+- Admin UI for provider setup and API key management
+- Health check endpoint for monitoring provider status
+- Enable/disable toggle for each provider
+- Graceful error handling with detailed logging
+
+### Supported Providers (7 platforms)
+- **Blazot**: Developer-focused social platform
+- **Bluesky**: AT Protocol integration
+- **Mastodon**: Federated network support
+- **Twitter/X**: OAuth-based integration
+- **TwitchChat**: Via relay server (see TwitchRelayUri config)
+- **YouTube**: Video hashtag search
+- **YouTube Live Chat**: Real-time chat integration
+- **Azure Queue**: Custom message delivery integration
+
+## Security Best Practices
+
+### Configuration Encryption
+- **Always encrypt** sensitive data (API keys, OAuth tokens, connection strings)
+- Uses **AES-256-CBC** encryption with configurable keys
+- See `doc/Configuration-Encryption.md` for implementation details
+- Encryption keys managed via environment variables or Azure Key Vault
+- Graceful degradation: app continues with warnings if encryption unavailable
+
+### API Keys and Secrets
+- **Never commit** API keys, tokens, or secrets to the repository
+- Use `.env.local` for local development (gitignored)
```

Comment:
Don't use .env.local - this is an Aspire managed application.  Use .NET user secrets with parameters in the AppHost project being passed into the child applications